networkit/cpp/graph/GraphTools.cpp

```diff
@@ -200,15 +200,66 @@ double density(const Graph &G) noexcept {
     return m / (n * (n - 1));
 }
 
-double volume(const Graph &G) {
-    if (G.isDirected())
-        throw std::runtime_error("Volume computation is only supported for undirected graphs.");
+double inVolume(const Graph &G, const std::unordered_set<node> &nodes) {
+    double inVolume = 0.0;
+    bool directed = G.isDirected() ? false : true;
+
+    // Compute volume of subgraph
+    if (!nodes.empty() && G.numberOfNodes() != nodes.size()) {
```

Comment:
While the optimization for not counting the whole graph seems neat, I wonder if we should not check that the graph really contains all nodes that are in the set.